pennylane/devices/default_qubit_torch.py

```diff
@@ -0,0 +1,350 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains a PyTorch implementation of the :class:`~.DefaultQubit`
+reference plugin.
+"""
+
+try:
+    import torch
+
+    v = torch.__version__.split(".")
+    if int(v[1]) < 8 or (int(v[1]) == 8 and int(v[2][0]) == 0):
+        raise ImportError("default.qubit.torch device requires Torch>=1.8.1")
+
+except ImportError as e:
+    raise ImportError("default.qubit.torch device requires Torch>=1.8.1") from e
+
+import warnings
+from string import ascii_letters as ABC
+import numpy as np
+from pennylane.operation import DiagonalOperation
+from pennylane.devices import torch_ops
+from . import DefaultQubit
+
+
+
+ABC_ARRAY = np.array(list(ABC))
+
+
+class DefaultQubitTorch(DefaultQubit):
+    """Simulator plugin based on ``"default.qubit"``, written using PyTorch.
+
+    **Short name:** ``default.qubit.torch``
+
+    This device provides a pure-state qubit simulator written using PyTorch.
+    As a result, it supports classical backpropagation as a means to compute the Jacobian. This can
+    be faster than the parameter-shift rule for analytic quantum gradients
+    when the number of parameters to be optimized is large.
+
+    To use this device, you will need to install PyTorch:
+
+    .. code-block:: console
+
+        pip install torch>=1.8.1
+
+    **Example**
+
+    The ``default.qubit.torch`` is designed to be used with end-to-end classical backpropagation
+    (``diff_method="backprop"``) with the PyTorch interface. This is the default method
+    of differentiation when creating a QNode with this device.
+
+    Using this method, the created QNode is a 'white-box', and is
+    tightly integrated with your PyTorch computation:
+
+    >>> dev = qml.device("default.qubit.torch", wires=1)
+    >>> @qml.qnode(dev, interface="torch", diff_method="backprop")
+    ... def circuit(x):
+    ...     qml.RX(x[1], wires=0)
+    ...     qml.Rot(x[0], x[1], x[2], wires=0)
+    ...     return qml.expval(qml.PauliZ(0))
+    >>> weights = torch.tensor([0.2, 0.5, 0.1], requires_grad=True)
+    >>> res = circuit(weights)
+    >>> res.backward()
+    >>> print(weights.grad)
+    tensor([-2.2527e-01, -1.0086e+00,  1.3878e-17])
```

Comment:
Might be prefered to use a code block here instead of a console, given the number of lines.
```suggestion
    .. code-block:: python
        dev = qml.device("default.qubit.torch", wires=1)
        @qml.qnode(dev, interface="torch", diff_method="backprop")
        def circuit(x):
            qml.RX(x[1], wires=0)
            qml.Rot(x[0], x[1], x[2], wires=0)
            return qml.expval(qml.PauliZ(0))
        weights = torch.tensor([0.2, 0.5, 0.1], requires_grad=True)
        res = circuit(weights)
        res.backward()
        
    >>> print(weights.grad)
    tensor([-2.2527e-01, -1.0086e+00,  1.3878e-17])
```

Reply:
Resolved.